tests/page-objects/references.ts

```diff
@@ -0,0 +1,441 @@
+      source: getContentEditableFieldByLabel('Source'),
+      version: this.page.locator('[name=version]'),
+      volume: this.page.locator('[name=volume]'),
+      firstPage: this.page.locator('[name=firstPage]'),
+      lastPage: this.page.locator('[name=lastPage]'),
+      eLocationId: this.page.locator('[name=elocationId]'),
+      doi: this.page.locator('[name=doi]'),
+      accession: this.page.locator('[name=accessionId]'),
+      website: getContentEditableFieldByLabel('Website'),
+      isbn: this.page.locator('[name=isbn]'),
+      url: this.page.locator('[name=extLink]'),
+      accessedDate: this.page.locator('[name=dateInCitation]'),
+      generatedOrAnalyzed: this.page.locator('#mui-component-select-specificUse'),
+      pmid: this.page.locator('[name=pmid]'),
+      inPress: this.page.locator('[name=inPress]'),
+      chapterTitle: getContentEditableFieldByLabel('Chapter title'),
+      bookTitle: getContentEditableFieldByLabel('Book title'),
+      edition: this.page.locator('[name=edition]'),
+      publisherLocation: this.page.locator('[name=publisherLocation]'),
+      publisherName: this.page.locator('[name=publisherName]'),
+      patentNumber: this.page.locator('[name=patent]'),
+      delete: this.page.locator('.MuiDialogContent-root').locator('text=DELETE'),
+      cancel: this.page.locator('text=CANCEL'),
+      done: this.page.locator('text=DONE'),
+    };
+  }
+
+  private async addAuthors(names: Names): Promise<void> {
+    for (let i = 0; i < names.length; i++) {
+      await this.modal.addAuthor.click();
+      await this.modal.authorLastName.nth(i).fill(names[i].lastName);
+      await this.modal.authorFirstName.nth(i).fill(names[i].firstName);
+    }
+  }
+
+  private static async assertNames(names: Names, ref: Locator): Promise<void> {
+    for (let i = 0; i < names.length; i++) {
+      const { firstName, lastName } = names[i];
+      await expect(ref).toContainText(`${lastName} ${firstName}`);
+    }
+  }
+
+  private async addEditors(names: Names): Promise<void> {
+    for (let i = 0; i < names.length; i++) {
+      await this.modal.addEditor.click();
+      await this.modal.editorLastName.last().fill(names[i].lastName);
+      await this.modal.editorFirstName.last().fill(names[i].firstName);
+    }
+  }
+
+  private async addReferenceAndSetCommonProperties(reference: Reference): Promise<void> {
+    await this.addReference.click();
+    await this.modal.type.click();
+    await this.page.locator(`text=${reference.type}`).last().click();
+    await this.addAuthors(reference.names);
+  }
+
+  private async addPageNumbersAndVolume(firstPage: number, lastPage: number, volume: string): Promise<void> {
+    await this.modal.firstPage.fill(String(firstPage));
+    await this.modal.lastPage.fill(String(lastPage));
+    await this.modal.volume.fill(volume);
+  }
+
+  private async setInPress(inPress?: boolean): Promise<void> {
+    if (inPress !== undefined) {
```

Comment:
I'd suggest using a default arg here rather than undefined, e.g.

```
  private async setInPress(inPress = true: boolean): Promise<void> {
    inPress ?  await this.modal.inPress.check() : await this.modal.inPress.uncheck();
  }
```